### ISSUE TITLE
How to finetune?

### ISSUE THREAD

**Issue by robotpp** [hasanirtiza/Pedestron#27]
Thanks for your work!!
I have read your pre-print version (Sec6.2) about how to achieve the best performance on Citypersons. However, when I use the pre-trained CrowdHuman model (AP84.2) and finetune on the Citypersons with the lr 0.0025 (1 img, 1 GPU), the 5 epoch result decreases a lot (just 84.4 MR-2 on Reasonable set).  
Can you share more details about how to finetune? Do I use the inappropriate finetune parameters or datasets?

**Comment by hasanirtiza** [comment]
We have tried multiple times (did it today as well) with the pre-trained model on CrowdHuman and fine-tuning it on CityPersons. Till now we got 9.56 on the reasonable set of CityPerson on the 3rd epoch (its still running). Since we used 7 gpus(2im/gpu) with learning rate 0.02. Learning rate you set also seems fine according the linear scaling rule. Maybe try 0.001 for starters. 
Additionally, what is the LAMR you get when you simply apply (without fine-tuning) on CityPersons the pre-trained model on CrowdHuman ?

**Comment by robotpp** [comment]
Thanks for your reply!!
The performance of the pre-trained CrowdHuman model (without fine-tuning on CityPersons) is really good (15.53 MR-2). The pre-trained CrowdHuman model is directly downloaded from the link of Google Drive you have provided. Can I directly use it to fine-tune on Citypersons? Or should I train the CrowdHuman model by myself and then fine-tune on Citypersons? I have also tried to use the ECP pre-trained model (also downloaded from the Google Drive) to fine-tune on Citypersons, and I face the same problem as CrowdHuman. I have tried learning rate from 0.0005 to 0.02, but the performance seems unsatisfactory (1st epoch nearly 100MR-2 on Citypersons).
I want to ask for advice:
1) can I directly download the pre-trained model on Google Drive and fine-tune on the other dataset?
2) will the training with the single GPU influence the performance during fine-tuning?
3) I just modify the Config file to use the pre-trained CrowdHuman model. Is it correct? For example,  I modify the file 'configs/elephant/cityperson/cascade_hrnet.py' pretrained='open-mmlab://msra/hrnetv2_w32' -> pretrained=path of the downloaded pre-trained model.
Thank you again for your reply!!!!

**Comment by hasanirtiza** [comment]
I think I know the problem (see point 3).  
1) Yes, you can and you should use the pre-trained model we provide for fine-tuning (that is why we provide it for general person detection).

2) It might effect but the effect should not be huge. I mean you can expect a performance drop of 1 or 2 MR-^2 points, but not what you are getting.

3) This is where the problem is, if you want to fine-tune using a pre-trained model, you do not change the pre-trained path, instead (undo it and use as provided in cofig file), in the config file, near the end you have a field `load_from`, which is set to `None` by default. Pass the model along with its path and you should be good to go.

`load_from = './models_pretrained/epoch_19.pth'`

P.S:- The code might complain that epoch_19.pth is missing (since you have epoch_19.pth.stu), simply make a copy of epoch_19.pth.stu as epoch_19.pth. 

This was also discuss here #17 near the end

**Comment by robotpp** [comment]
It really works!!! Thank you very much!! :)
I have fine-tuned on the Citypersons with the pre-trained CrowdHuman model (AP84.2), and I got 9.25 on the Reasonable set (2nd epoch) with lr 0.001 (1 img, 1GPU).
The problem before was the incorrect manner to load the pre-trained model.

**Comment by lazrak-mouad** [comment]
> It really works!!! Thank you very much!! :)
> I have fine-tuned on the Citypersons with the pre-trained CrowdHuman model (AP84.2), and I got 9.25 on the Reasonable set (2nd epoch) with lr 0.001 (1 img, 1GPU).
> The problem before was the incorrect manner to load the pre-trained model.

Would you kindly share your full command used for this, because I'm struggling with almost the same issue.

Thank you.

**Comment by robotpp** [comment]
> > It really works!!! Thank you very much!! :)
> > I have fine-tuned on the Citypersons with the pre-trained CrowdHuman model (AP84.2), and I got 9.25 on the Reasonable set (2nd epoch) with lr 0.001 (1 img, 1GPU).
> > The problem before was the incorrect manner to load the pre-trained model.
> 
> Would you kindly share your full command used for this, because I'm struggling with almost the same issue.
> 
> Thank you.

I just modify the config file "load_from = None -> load_from = path of the pre-trained model".
